examples/gno.land/p/demo/acl/acl.gno

```diff
@@ -71,17 +82,28 @@ func (d *Directory) AddGroupPerm(name string, verb, resource string) {
 		verbs:     []string{verb},
 		resources: []string{resource},
 	}
-	d.addPermToBucket(bucket, p)
+	d.addPermsToBucket(bucket, []perm{p})
+}
+
+func (d *Directory) AddGroupPerms(name string, verbs []string, resource string) {
+	bucket := "g:" + name
+	for _, verb := range verbs {
+		p := perm{
+			verbs:     []string{verb},
+			resources: []string{resource},
+		}
+		d.addPermsToBucket(bucket, []perm{p})
+	}
 }
 
-func (d *Directory) addPermToBucket(bucket string, p perm) {
+func (d *Directory) addPermsToBucket(bucket string, p []perm) {
 	var ps perms
 
 	existing, ok := d.permBuckets.Get(bucket)
 	if ok {
 		ps = existing.(perms)
 	}
-	ps = append(ps, p)
+	ps = append(ps, p...)
```

Comment:
Since there is no duplication check in here, it is possible for duplicate permission to be added. For example, if the same permission is added multiple times and then removed, only one instance will be deleted rather than all of them. This might be lead to unexpected behaviour.

```go
func TestPermDuplication(t *testing.T) {
	dir := New()
	dir.AddUserPerm(testutils.TestAddress("user"), "write", "r/demo/boards:gnolang/3")
	dir.AddUserPerm(testutils.TestAddress("user"), "write", "r/demo/boards:gnolang/3")

	shouldHasPerm(t, dir, testutils.TestAddress("user"), "write", "r/demo/boards:gnolang/3")

	dir.RemoveUserPerm(testutils.TestAddress("user"), "write", "r/demo/boards:gnolang/3")
	shouldNotHasPerm(t, dir, testutils.TestAddress("user"), "write", "r/demo/boards:gnolang/3")
}
```

test result
```plain
=== RUN   TestPermDuplication
uassert.Equal: same type but different value
        expected: false
        actual:   true - g1w4ek2ujlta047h6lta047h6lta047h6lkvk347 should not has perm for write - r/demo/boards:gnolang/3
--- FAIL: TestPermDuplication (0.00s)
```

Reply:
https://github.com/gnolang/gno/pull/2987/commits/f547dd03a6db2a7d1c4c85caa6a6f315799d78e0